pkg/controller/reconciler.go

```diff
@@ -68,19 +68,35 @@ func (o *operationAnnotationWrapper) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	if ok, err := o.removeOperationAnnotation(obj, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile, v1beta1constants.GardenerOperationRestore); !ok {
+		return reconcile.Result{}, err
+	}
+
+	return o.Reconciler.Reconcile(request)
+}
+
+func (o *operationAnnotationWrapper) removeOperationAnnotation(obj runtime.Object, key string, values ...string) (bool, error) {
```

Comment:
Not sure that it is necessary to return (bool, err) here. Anytime err != nil is returned, false is also returned, and when err == nil, trrue is returned. You could just reuse the err.

Reply:
You are right. Thanks

Reply:
The function still returns err and bool :)

Reply:
Not enymore